pkg/task/storage.go

```diff
@@ -0,0 +1,155 @@
+package task
+
+import (
+	"container/heap"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+// TaskStorage is a leveldb-backed priority queue.
+// storage is persisted before the in-memory queue to prevent inconsistency between restarts; what
+// you see in the queue is the same as what is written the database
+type TaskStorage struct {
+	Max       int
+	Path      string
```

Comment:
Users probably shouldn't be setting these values on the struct directly. We probably want a constructor:

```go
path := filepath.Join(envconfig.Dirs().Work(), "tasks.db")
tqueue, err := taskqueue.Open(path)
```

Reply:
Changed to a New-* style constructor.